controllers/nginx/pkg/config/config.go

```diff
@@ -307,20 +301,14 @@ func NewDefault() Configuration {
 	return cfg
 }
 
-// BuildLogFormatUpstream format the log_format upstream based on proxy_protocol
-func BuildLogFormatUpstream(useProxyProtocol bool, curLogFormatUpstream string) string {
-
-	// test if log_format comes from configmap
-	if curLogFormatUpstream != "" &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr") &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$remote_addr") {
-		return curLogFormatUpstream
-	}
-
-	if useProxyProtocol {
-		return fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr")
+// BuildLogFormatUpstream format the log_format upstream using
+// proxy_protocol_addr as remote client address if UseProxyProtocol
+// is enabled.
+func (cfg Configuration) BuildLogFormatUpstream() string {
```

Comment:
@aledbf Don't forget to compare if the logFormatUpstream is configured via configmap. (--log-format-upstream)